### DISCUSSION TITLE
Calling Powershell scripts from MQL?

### DISCUSSION THREAD

**Discussion by GurdipS5** [mondoohq/cnquery#1096] [Q&A] [answered]
Hi,
I have just come across Mondoo and it looks great. I saw I can write custom checks with Powershell (or rather, MQL that calls Powershell). Is it possible, however, for the MQL to call an external Powershell script?

My infra is on prem Windows servers.

**Comment by chris-rock** [comment]
Welcome @GurdipS5

Great question. You can test then when you run the `cnspec shell`. When you embed the powershell code, it looks like that:

```
cnspec> powershell("Write-Host 'Hello, World!'").stdout
powershell.stdout: "Hello, World!
"
```

Assuming you have a local file `test.ps1`

```powershell
Write-Host 'Hello, World!'
```

Then you can just call the script

```
cnspec> powershell("./test.ps1").stdout
powershell.stdout: "Hello, World!
"
```

For production use I would recommend using absolute paths. In the second case the powershell script is not distributed via the policy itself and you would need to make sure the powershell script is available at all machines.

**Comment by GurdipS5** [comment]
Yeah I will be testing this. Just discovered Mondoo late last night lol.
Great tool there aren't many policy as code tools for Windows I was
thinking Pester.

**Reply by GurdipS5** [reply]
And the scripts will be distributed via Octopus Deploy and in git so that will contrl that.

**Reply by chris-rock** [reply]
Nice

**Comment by chris-rock** [comment, marked as answer]
We have a lot of Windows users using cnspec in production to secure their Windows server and Windows client fleet. Here is our Windows Security policy https://github.com/mondoohq/cnspec-policies/blob/main/core/mondoo-windows-security.mql.yaml and our Windows Workstation Security Policy https://github.com/mondoohq/cnspec-policies/blob/main/core/mondoo-windows-workstation-security.mql.yaml

We also have support for all CIS Benchmarks if you need that.

**Comment by GurdipS5** [comment]
Awesome. I'd add I'm using iis so that's a focus. Server 2022.
